### ISSUE TITLE
Beacons cannot be reloaded/refreshed without minion restart

### ISSUE THREAD

**Issue by basepi** [saltstack/salt#35960]
### Description of Issue/Question

If I sync changes to an existing custom beacon via `_beacons/` and `saltutil.sync_all`, the new changes are not picked up by the minion unless I restart the minion. `saltutil.refresh_beacons` also does not work. The event fires, but the new beacon is not loaded.
### Steps to Reproduce Issue
- Sync a custom beacon to the minion, with some logs so you can see it firing.
- Add new logs to the custom beacon
- Sync the new version with `saltutil.sync_all` (and `saltutil.refresh_beacons` for good measure)
- Watch for the new logs to appear (they won't)
### Versions Report

```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.21.0
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: 0.21.4
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.18.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

**Comment by Ch3LL** [comment]
@basepi thanks for the heads up. We definitely need to make sure these are refreshed without requiring a restart.

I'm really not certain on this one but it seems like when it attempts to refresh the beacons it fires and event. And I think the event gets handled [here](https://github.com/saltstack/salt/blob/develop/salt/minion.py#L1831)  Well there is a `module_refresh` and a `pillar_refresh` but not a `beacons_refresh` which is the event generated for the beacon refresh method in saltutil.

**Comment by basepi** [comment]
Yeah, I saw the event fire as well, but it obviously wasn't being handled correctly. I will probably investigate this myself when I find time, unless someone over there gets to it. (Not like you guys are busy, right??? ;) )

**Comment by jagguli** [comment]
Pull request with fix I'm testing my self ATM https://github.com/saltstack/salt/pull/40892

**Comment by golmaal** [comment]
@Ch3LL @garethgreenaway - I would like to request you to consider merging the fix for this issue in next 2017.7.x release.
